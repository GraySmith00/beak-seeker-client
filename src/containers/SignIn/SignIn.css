.sign-in {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  margin: 0 auto;
  position: relative;
  background: url("./birdsSunset1.jpg");
  background-size: cover;
  background-repeat: no-repeat;
  background-position: bottom center;
  width: 100%; }
  .sign-in .welcome-text {
    width: 100%;
    margin: 0;
    padding-top: 40%;
    color: #fff; }
  .sign-in .fa-dove {
    margin-right: 1rem; }
  .sign-in h3 {
    margin: 0;
    font-size: 0.8rem; }
  .sign-in h2 {
    margin-top: 0;
    font-size: 2.5rem;
    font-family: 'Pacifico', cursive; }
  .sign-in .app-description {
    position: absolute;
    top: 30%;
    text-align: center;
    background-color: rgba(0, 0, 0, 0.75);
    width: 75%;
    padding: 1rem;
    border-radius: 5px;
    color: #fff; }
  .sign-in .twitter-button {
    padding: 1rem;
    margin: 5rem auto;
    text-decoration: none;
    width: 75%;
    text-align: center;
    border-radius: 5px;
    color: #fff;
    background: #00c6ff;
    /* fallback for old browsers */
    background: -webkit-linear-gradient(to right, #0072ff, #00c6ff);
    /* Chrome 10-25, Safari 5.1-6 */
    background: linear-gradient(to right, #0072ff, #00c6ff);
    /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
    -webkit-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43); }
