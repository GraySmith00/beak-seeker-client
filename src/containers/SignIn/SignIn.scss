.sign-in {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  margin: 0 auto;
  position: relative;

  .hero {
    background: url('./birdsSunset1.jpg');
    background-size: cover;
    background-repeat: no-repeat;
    background-position: bottom center;
    height: 300px;
    width: 100%;

    -webkit-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
  }

  .overlay {
    background-color: rgba(0, 0, 0, 0.3);
    height: 100%;
    width: 100%;
  }

  .welcome-text {
    width: 83%;

    margin: 0 auto;
    padding-top: 20%;
    color: #fff;
  }

  .fa-dove {
    margin-right: 1rem;
  }

  h3 {
    margin: 0;
    font-size: 0.8rem;
  }

  h2 {
    margin-top: 0.5rem;
    font-size: 2.5rem;
    font-family: 'Pacifico', cursive;
  }

  .app-description {
    position: absolute;
    top: 42%;
    text-align: center;
    background-color: #1b2026;
    width: 75%;
    padding: 1rem;
    border-radius: 5px;
    color: #fff;
  }

  .twitter-button {
    padding: 1rem;
    margin: 3rem auto;
    text-decoration: none;

    border-radius: 5px;
    color: #fff;

    background: #00c6ff; /* fallback for old browsers */
    background: -webkit-linear-gradient(
      to right,
      #0072ff,
      #00c6ff
    ); /* Chrome 10-25, Safari 5.1-6 */
    background: linear-gradient(
      to right,
      #0072ff,
      #00c6ff
    ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */

    -webkit-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 43px -8px rgba(0, 0, 0, 0.43);
  }
}
