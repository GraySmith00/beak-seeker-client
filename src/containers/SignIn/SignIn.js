import React, { Component } from 'react';

import './SignIn.css';

class SignIn extends Component {
  render() {
    return (
      <div className="sign-in">
        <div className="hero">
          <div className="overlay">
            <div className="welcome-text">
              <h3>Welcome to</h3>
              <h2>
                <i className="fas fa-dove" />
                BeakSeeker
              </h2>
            </div>
          </div>
        </div>
        <div className="app-description">
          <p>
            BeakSeeker is the most fun way to go birding! Find the best local
            hotspots. See which birds have been recently seen in your area. Earn
            badges for spotting birds and compete against your fellow birders to
            see who is the ultimate beak seeker!
          </p>
        </div>
        <a
          href="https://gs-beakseeker-server.herokuapp.com/twitter/login"
          className="twitter-button"
        >
          Sign in with Twitter
        </a>
      </div>
    );
  }
}

export default SignIn;
