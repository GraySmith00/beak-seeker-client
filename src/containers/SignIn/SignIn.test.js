import React from 'react';
import { shallow } from 'enzyme';

import SignIn from './SignIn';

describe('SignIn component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignIn />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
