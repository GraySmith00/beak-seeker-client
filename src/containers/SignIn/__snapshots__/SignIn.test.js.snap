// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="hero"
>
          <div
                    className="overlay"
          >
                    <div
                              className="welcome-text"
                    >
                              <h3>
                                        Welcome to
                              </h3>
                              <h2>
                                        <i
                                                  className="fas fa-dove"
                                        />
                                        BeakSeeker
                              </h2>
                    </div>
          </div>
</div>,
        <div
          className="app-description"
>
          <p>
                    Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.
          </p>
</div>,
        <a
          className="twitter-button"
          href="http://localhost:5000/twitter/login"
>
          Sign in with Twitter
</a>,
      ],
      "className": "sign-in",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="overlay"
>
            <div
                        className="welcome-text"
            >
                        <h3>
                                    Welcome to
                        </h3>
                        <h2>
                                    <i
                                                className="fas fa-dove"
                                    />
                                    BeakSeeker
                        </h2>
            </div>
</div>,
          "className": "hero",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="welcome-text"
>
              <h3>
                            Welcome to
              </h3>
              <h2>
                            <i
                                          className="fas fa-dove"
                            />
                            BeakSeeker
              </h2>
</div>,
            "className": "overlay",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Welcome to
</h3>,
                <h2>
                  <i
                                    className="fas fa-dove"
                  />
                  BeakSeeker
</h2>,
              ],
              "className": "welcome-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome to",
                },
                "ref": null,
                "rendered": "Welcome to",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fas fa-dove"
/>,
                    "BeakSeeker",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-dove",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "BeakSeeker",
                ],
                "type": "h2",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.
</p>,
          "className": "app-description",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.",
          },
          "ref": null,
          "rendered": "Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign in with Twitter",
          "className": "twitter-button",
          "href": "http://localhost:5000/twitter/login",
        },
        "ref": null,
        "rendered": "Sign in with Twitter",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="hero"
>
            <div
                        className="overlay"
            >
                        <div
                                    className="welcome-text"
                        >
                                    <h3>
                                                Welcome to
                                    </h3>
                                    <h2>
                                                <i
                                                            className="fas fa-dove"
                                                />
                                                BeakSeeker
                                    </h2>
                        </div>
            </div>
</div>,
          <div
            className="app-description"
>
            <p>
                        Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.
            </p>
</div>,
          <a
            className="twitter-button"
            href="http://localhost:5000/twitter/login"
>
            Sign in with Twitter
</a>,
        ],
        "className": "sign-in",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="overlay"
>
              <div
                            className="welcome-text"
              >
                            <h3>
                                          Welcome to
                            </h3>
                            <h2>
                                          <i
                                                        className="fas fa-dove"
                                          />
                                          BeakSeeker
                            </h2>
              </div>
</div>,
            "className": "hero",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="welcome-text"
>
                <h3>
                                Welcome to
                </h3>
                <h2>
                                <i
                                                className="fas fa-dove"
                                />
                                BeakSeeker
                </h2>
</div>,
              "className": "overlay",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Welcome to
</h3>,
                  <h2>
                    <i
                                        className="fas fa-dove"
                    />
                    BeakSeeker
</h2>,
                ],
                "className": "welcome-text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Welcome to",
                  },
                  "ref": null,
                  "rendered": "Welcome to",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fas fa-dove"
/>,
                      "BeakSeeker",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fas fa-dove",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "BeakSeeker",
                  ],
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.
</p>,
            "className": "app-description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.",
            },
            "ref": null,
            "rendered": "Readymade literally sartorial keytar narwhal, tofu everyday carry fingerstache vice whatever hashtag put a bird on it. Gastropub poutine helvetica, lyft letterpress hexagon slow-carb XOXO. Migas fixie lomo, swag stumptown +1 put a bird on it wolf tilde PBR&B try-hard subway tile chicharrones typewriter.",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign in with Twitter",
            "className": "twitter-button",
            "href": "http://localhost:5000/twitter/login",
          },
          "ref": null,
          "rendered": "Sign in with Twitter",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
